### PR TITLE
User tags

### PR DESCRIPTION
**Pull request by bwozniak27**
User search, login page redirect, and 404 page

## Description
I created a search bar component and added regex searching as the user types to filter the users that display in the table. Users can be found by searching for first name, last name, username, or email. For the routing, I passed UserContext to every page and checked if the user was logged in or not. If they were not, it would redirect them to the login page. Additionally, if a user is logged in, if they access the login page it redirects them to report-list. Then finally, I added a 404 page that will appear if a user attempts to go anywhere that doesn't exist.

## Related Monday.com Ticket
- Search and Filtering Viewers page
- Auto-redirect to login page for non authenticated users
- 404 page

## Related Issue
- fixes #

## How Has This Been Tested?
- A variety of searches were ran to identify users. 
- For the login, I logged out, then attempted to access pages by manually typing in url. Additionally, as a logged in user, I attempted to access the login page.

Notes:
- upon crash (force killing application), user remains logged in. I don't think its an issue, but I could be wrong
- Currently, the redirects are done by including a redirect in each page. This will require us to include it for every page added. I attempted to create a parent component to make a global redirect, but was unsuccessful.

### CODE REVIEW

File: meteor-app/imports/ui/Viewers_List/Viewers_List.tsx [hunk 1]
```diff
@@ -15,11 +17,39 @@ const usePage = (account_id) => useTracker(() => {
...
+
+	useEffect(() => {
+		filterDishes()
+	}, [query])
+
+	useEffect(() => {
+		setViewersShown(viewers)
+	}, [viewers])
+
+	const filterDishes = () => {
```
Comment (NathanJeanDevs) [inline meteor-app/imports/ui/Viewers_List/Viewers_List.tsx]: What is "Dishes" ?

**Review by NathanJeanDevs — CHANGES_REQUESTED** [review]
This looks to be the same as PR #14 

Can you please clarify what is different in this PR?

File: meteor-app/imports/ui/Viewers_List/Viewers_List.tsx (hunk 1 above)
Comment (bwozniak27) [reply meteor-app/imports/ui/Viewers_List/Viewers_List.tsx]: Oh I found that function somewhere and modified it, forgot to change the function name

**Comment by bwozniak27** [comment]
I branched off this branch for the styling PR, so the changes in here are also in that PR

**Pull request closed without merging**